backend-tests/tests/test_deployments.py

```diff
@@ -0,0 +1,257 @@
+                             timedelta(days=1)).isoformat("T")) + "Z",
+                        "batch_size":
+                        10,
+                    },
+                    {
+                        "start_ts":
+                        str((datetime.utcnow() +
+                             timedelta(days=1)).isoformat("T")) + "Z",
+                        "batch_size":
+                        10,
+                    },
+                    {
+                        "start_ts":
+                        str((datetime.utcnow() +
+                             timedelta(days=1)).isoformat("T")) + "Z",
+                        "batch_size":
+                        10,
+                    },
+                    {
+                        "start_ts":
+                        str((datetime.utcnow() +
+                             timedelta(days=2)).isoformat("T")) + "Z",
+                        # Batch size is optional in the last stage (ie, it is the remaining devices)
+                    },
+                ],
+            },
+            {
+                "name":
+                "fourth-test-case",
+                "artifact_name":
+                "deployments-phase-testing",
+                "device_count":
+                100,
+                "phases": [{
+                    "batch_size": 10,
+                    "device_count": 0,
+                }, {
+                    "batch_size": 10,
+                    "device_count": 0,
+                }, {
+                    "batch_size": 10,
+                    "device_count": 0,
+                }, {
+                    "batch_size": 10,
+                    "device_count": 0,
+                }, {
+                    "batch_size": 60,
+                    "device_count": 0,
+                }],
+            }),
+    ]
+
+    @pytest.mark.parametrize("test_case", test_cases)
+    def test_phased_deployments_success(self, test_case,
+                                        setup_deployments_enterprise_test):
+        user, tenant, utoken, devices = setup_deployments_enterprise_test
+        request_body, expected_response_json = test_case
+        deploymentclient = ApiClient(
+            api.client.GATEWAY_URL + "/api/management/v1/deployments")
+        # Set the device pool in the request body
+        request_body["devices"] = [device["id"] for device in devices]
+        resp = deploymentclient.with_auth(utoken).call(
+            'POST',
+            '/deployments',
+            body=request_body,
+        )
+        assert resp.status_code == 201
+        # Store the location for the GET /deployments/{id} request
+        deployment_id = os.path.basename(resp.headers['Location'])
+        # GET /deployments (unfiltered)
+        resp = deploymentclient.with_auth(utoken).call('GET', '/deployments')
+        assert resp.status_code == 200
+        # POST /deployments and GET /deployments should yield equal results
+        response_body_dict = resp.json()[0]
```

Comment:
I know at least for now it's just one deployment (but I'm mentioning >1 in other comments) - but never rely on the returned order moving forward. just a note.